mxs: fix Duckbill SD card image generation and whitespace fixes

mhei opened the pull request:
This PR includes two patches for the mxs target:
- the first one fixes the image generation for I2SE Duckbill - which slightly differs from the Olinuxino boards
- the second one is only a whitespace fix - house-keeping keep the target in sync with others

mhei committed:
mxs: fix image generation for I2SE Duckbills

The standard U-Boot boot scripts for Duckbills expect the
Linux kernel and device tree files installed below /boot
within the (ext4) root filesystem.
Also a raw zImage is expected instead of uImage.

Extend the SD card generation accordingly and while at,
install all possible Duckbill DT blobs there.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>

mhei committed:
mxs: image: slightly adjust whitespace (no functional change)

Seems that the common style is to have whitespace around the operators
and the indent is single tab.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>

wigyori commented:
LGTM (although no Duckbill hardware to test on), but with the apk switch in trunk, you'll need this as well (or switch the package deselect to nftables, whichever you prefer):

```
diff --git a/target/linux/mxs/image/Makefile b/target/linux/mxs/image/Makefile
index 0281600da0..499dc5e75c 100644
--- a/target/linux/mxs/image/Makefile
+++ b/target/linux/mxs/image/Makefile
@@ -68,8 +68,7 @@ endef
 define Device/i2se_duckbill
   DEVICE_VENDOR := I2SE
   DEVICE_MODEL := Duckbill
-  DEVICE_PACKAGES := -dnsmasq -ppp -ip6tables -iptables -mtd \
-       uboot-envtools kmod-leds-gpio -kmod-nf-nathelper
+  DEVICE_PACKAGES := -dnsmasq -ppp -mtd uboot-envtools kmod-leds-gpio
   SUPPORTED_DEVICES := i2se,duckbill
   SOC := imx28
   KERNEL := kernel-bin
```
... otherwise apk will stop with:

```
IPKG_INSTROOT=/u/owrt4/mxs/build_dir/target-arm_arm926ej-s_musl_eabi/linux-mxs_generic/target-dir-003f2faa /u/owrt4/mxs/staging_dir/host/bin/fakeroot /u/owrt4/mxs/staging_dir/host/bin/apk --root /u/owrt4/mxs/build_dir/target-arm_arm926ej-s_musl_eabi/linux-mxs_generic/target-dir-003f2faa --keys-dir /u/owrt4/mxs --no-logfile --preserve-env --no-scripts --repositories-file /dev/null --repository file:///u/owrt4/mxs/staging_dir/packages/mxs/packages.adb del dnsmasq ppp ip6tables iptables mtd kmod-nf-nathelper
ERROR: No such package: ip6tables
ERROR: No such package: iptables
ERROR: No such package: kmod-nf-nathelper
```

If you agree I'll add the patch above when merging the PR.

mhei committed:
mxs: adapt default package list for Duckbill devices

Replace outdated package names, order the remaining.

firewall4 is added again, since it is enabled by default.
The device is not a router by default, so this package
makes no sense in the default config. User can enable it
by theirself, or it will be automatically pulled via
dependency, e.g. luci-app-firewall.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>

mhei commented:
Thanks for review and the hint for apk. I've adapted the list and this works for me with latest trunk. Or is there any other branch regarding apk which I shall test against?

blogic commented:
@wigyori are you ok with the latest changes ?

wigyori commented:
Merged to trunk - thanks @mhei .
